SU2_CFD/src/numerics/viscous/CAvgGradCorrected_P1.cpp

```diff
@@ -0,0 +1,109 @@
+/*!
+ * \file numerics_direct_radiation.cpp
+ * \brief Numerical methods for computing the viscous residual in the P1 equation.
+ * \author Ruben Sanchez
+ * \version 7.0.1 "Blackbird"
+ *
+ * SU2 Project Website: https://su2code.github.io
+ *
+ * The SU2 Project is maintained by the SU2 Foundation
+ * (http://su2foundation.org)
+ *
+ * Copyright 2012-2019, SU2 Contributors (cf. AUTHORS.md)
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "../../../include/numerics/CNumericsRadiation.hpp"
+#include "../../../include/numerics/viscous/CAvgGradCorrected_P1.hpp"
+
+CAvgGradCorrected_P1::CAvgGradCorrected_P1(unsigned short val_nDim, unsigned short val_nVar,
+                                                   CConfig *config) : CNumericsRadiation(val_nDim, val_nVar, config) {
+
+  unsigned short iVar;
+
+  GammaP1 = 1.0 / (3.0*(Absorption_Coeff + Scattering_Coeff));
+
+  Edge_Vector = new su2double [nDim];
+  Proj_Mean_GradP1Var_Edge = new su2double [nVar];
+  Proj_Mean_GradP1Var_Kappa = new su2double [nVar];
+  Proj_Mean_GradP1Var_Corrected = new su2double [nVar];
+  Mean_GradP1Var = new su2double* [nVar];
+  for (iVar = 0; iVar < nVar; iVar++)
+    Mean_GradP1Var[iVar] = new su2double [nDim];
+
+}
+
+CAvgGradCorrected_P1::~CAvgGradCorrected_P1(void) {
+
+  delete [] Edge_Vector;
+  delete [] Proj_Mean_GradP1Var_Edge;
+  delete [] Proj_Mean_GradP1Var_Kappa;
+  delete [] Proj_Mean_GradP1Var_Corrected;
+  for (unsigned short iVar = 0; iVar < nVar; iVar++)
+    delete [] Mean_GradP1Var[iVar];
+  delete [] Mean_GradP1Var;
```

Comment:
`if (Mean_GradP1Var != nullptr)` around the whole thing and in the declaration of these member variables you default initialize them to nullptr.
But like I said it is not likely that these classes will get default constructed and I probably forgot these checks in some of the classes I changed too xD